src/components/applications/ApplicationsView.js

```diff
@@ -26,6 +30,22 @@ export const ApplicationsView = ({getApplications, isHR, isDev, mailInBase64, or
     const [reload, setReload] = useState(false)
     const [selectedApplicationNotes, setSelectedApplicationNotes] = useState([])
 
+    // state for filter and sort
+    const [fixedApplications, setFixedApplications] = useState([])
+    const [sort, setSort] = useState({apply: (applications) => applications})
+
+    const handleFilterSubmitted = (filters) => {
+        if(areFiltersValid(filters)){
+            setApplications(sort.apply(filterItems(fixedApplications, filters)))
+        }
+        else {
+            setApplications(sort.apply(fixedApplications))
+        }
+    }
+
+    const areFiltersValid = (filters) => filters.map(filter => filter.value)
+        .some(value => value?.length !== undefined ? value?.length > 0 : !!value)
```

Comment:
Może głupie pytanie length jako takie może kiedyś być undefinied?

Reply:
Jeśli to nie będzie tablica. Zabezpieczenie na wypadek, gdybyśmy dodali jakiś inny filter, który nie jest tablicą. A potrzebny taki workaround, bo !![] (czyli pusta tablica zcastowana do wartości logicznej) to true. Nawet jeśli porównujesz [] === true